pkg/util/sort_uniq_test.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package util
+
+import (
+	"fmt"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSortUniqInPlace(t *testing.T) {
+	elements := []string{"tag2:tagval", "tag1:tagval", "tag2:tagval"}
+	elements = SortUniqInPlace(elements)
+
+	assert.ElementsMatch(t, elements, []string{"tag1:tagval", "tag2:tagval"})
+}
+
+func benchmarkDeduplicateTags(b *testing.B, numberOfTags int) {
+	tags := make([]string, 0, numberOfTags+1)
+	for i := 0; i < numberOfTags; i++ {
+		tags = append(tags, fmt.Sprintf("aveeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeerylong:tag%d", i))
+	}
+	// this is the worst case for the insertion sort we are using
+	sort.Sort(sort.Reverse(sort.StringSlice(tags)))
+
+	tempTags := make([]string, len(tags))
+	copy(tempTags, tags)
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	for n := 0; n < b.N; n++ {
+		SortUniqInPlace(tempTags)
```

Comment:
As you sort several times the same reference, the benchmark for insertion sort is not relevant.